egui/src/widgets/plot/mod.rs

```diff
@@ -339,22 +340,36 @@ impl Widget for Plot {
         } = self;
 
         let plot_id = ui.make_persistent_id(id_source);
-        let memory = ui
+        let mut memory = ui
             .memory()
             .id_data
             .get_mut_or_insert_with(plot_id, || PlotMemory {
                 bounds: min_auto_bounds,
                 auto_bounds: !min_auto_bounds.is_valid(),
                 hovered_entry: None,
                 hidden_items: HashSet::new(),
+                min_auto_bounds,
             })
             .clone();
 
+        // If the min bounds changed, recalculate everything.
+        if min_auto_bounds != memory.min_auto_bounds {
+            memory = PlotMemory {
+                bounds: min_auto_bounds,
```

Comment:
Would this make sense:

```suggestion
                bounds: memory.bounds.merge(min_auto_bounds),
```

That is, if the new `min_auto_bounds` are larger, we zoom out to include them, but otherwise we keep the same bounds.

Reply:
Hmmmm I can see how this approach might make sense as a solution for the behaviour of the `include_*` methods. However, for our case, we want it so that if we decrease the bounds, it appears as though we are "zooming" into the plot. That is, we need the ability to specify the exact bounds of the visible area of the plot each update - if not via the `include_*` methods, perhaps some other method.